chromium_src/net/cookies/cookie_monster.cc

```diff
@@ -0,0 +1,136 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "net/cookies/cookie_monster.h"
+
+#include <iostream>
+#include <memory>
+#include "net/base/url_util.h"
+
+#define CookieMonster SimpleCookieMonster
+#include "../../../../net/cookies/cookie_monster.cc"
+#undef CookieMonster
+
+namespace net {
+
+namespace {
+
+CookieOptions OptionsWithoutEphemeralStorageURL(const CookieOptions& options) {
+  CookieOptions new_options(options);
+  new_options.top_frame_url_ = GURL();
+  return new_options;
+}
+
+}  // namespace
+
+CookieMonster::CookieMonster(scoped_refptr<PersistentCookieStore> store,
+                             NetLog* net_log)
+    : SimpleCookieMonster(store, net_log),
+      net_log_(
+          NetLogWithSource::Make(net_log, NetLogSourceType::COOKIE_STORE)) {}
+
+CookieMonster::CookieMonster(scoped_refptr<PersistentCookieStore> store,
+                             base::TimeDelta last_access_threshold,
+                             NetLog* net_log)
+    : SimpleCookieMonster(store, last_access_threshold, net_log),
+      net_log_(
+          NetLogWithSource::Make(net_log, NetLogSourceType::COOKIE_STORE)) {}
+
+CookieMonster::~CookieMonster() {}
+
+SimpleCookieMonster*
+CookieMonster::GetOrCreateEphemeralCookieStoreForTopFrameURL(
+    const GURL& top_frame_url) {
+  std::string domain = URLToEphemeralStorageDomain(top_frame_url);
+  auto it = ephemeral_cookie_stores_.find(domain);
```

Comment:
no need for this check, `emplace` will do the trick on its own

Reply:
Won't this result in creating another instantiation of `ChromiumCookieMonster` every time we call this method?

Reply:
yeah, it seems we should leave it alone